Navigation MPC

Runway27 opened the pull request:
## Main Changes

The first GNSS location is set as the initial state. The target location is user input. The MPC is run using an OSQP solver. It regularly checks whether the trajectory is correct with the latest GNSS location. 



## Tests

Yet to be implemented and tested

Runway27 committed:
Create Nav_MPC.py

Need to integrate with the GPS

Runway27 committed:
Update Nav_MPC.py

Uses GNSS locations now

alphaville commented on raspberry/bzzz/controllers/Nav_MPC.py:
```diff
@@ -0,0 +1,217 @@
+from threading import Thread, Lock
+from gnss import Gnss  
+
+class MPCController:
```
Rename to something like `NavigationMPC`

alphaville replied:
Add documentation

Runway27 committed:
Update Nav_MPC.py

The code is now in a blocking format than in a threading format

Runway27 committed:
Update main.py

The main file has an object for NAV_MPC. And the NAV_MPC is now being called in the main file

Runway27 committed:
Create Simulations

Runway27 committed:
Update Simulations

MPC simulations

Runway27 committed:
Create simulations

alphaville commented:
Closing due to inactivity
